src/randoop/ExceptionCheck.java

```diff
@@ -0,0 +1,157 @@
+package randoop;
+
+import java.util.Objects;
+
+/**
+ * An {@code ExceptionCheck} is used to indicate that an exception is expected
+ * at a particular statement in a sequence. Depending on command-line arguments
+ * to Randoop, an instance may be either a {@link ExpectedExceptionCheck} or 
+ * {@link EmptyExceptionCheck}. Both will catch the exception if it occurs, but
+ * differ on whether the expectation of the exception is enforced.
+ */
+public abstract class ExceptionCheck implements Check {
+  
+  private static final long serialVersionUID = 4806179088639914364L;
+
+  protected final Throwable exception;
+  
+  // Indicates which statement results in the given exception. 
+  protected final int statementIndex;
+
+  private String catchClassName;
+
+  /**
+   * Creates an exception check for the statement at the statement index.
+   * The generated code for this check will include a try-catch block with
+   * behaviors determined by implementing sub-classes. The catch class name
```

Comment:
I might drop this last sentence.  It's true but is a bit of an implementation detail and might be confusing to someone who is just trying to get an understanding of the class.  I had to read it a couple of times even though we just talked about this issue.